treeherder/perf/alerts.py

```diff
@@ -120,3 +123,125 @@ def generate_new_alerts_in_series(signature):
+        The returned list must contain regressions and (if present) improvements.
+        '''
+        regressions_only = all(alert.is_regression for alert in sorted_alerts)
+        improvements_only = all(not alert.is_regression for alert in sorted_alerts)
+
+        if regressions_only or improvements_only:
+            resulted_alerts_list = self._ensure_platform_variety(sorted_alerts)
+        else:  # mixed alert types
+            regressions = [alert for alert in sorted_alerts if alert.is_regression]
+            improvements = [alert for alert in sorted_alerts if not alert.is_regression]
+
+            if len(regressions) > 1:
+                regressions = self._ensure_platform_variety(regressions)
+                regressions[-1] = improvements[0]
+            regressions.append(improvements[0])
+            resulted_alerts_list = regressions
+
+        return resulted_alerts_list[
+               :self.max_improvements if improvements_only else self.max_alerts
+               ]
+
+    def _ensure_platform_variety(self, sorted_all_alerts: List[PerformanceAlert]):
+        '''
+        Note: Ensure that the sorted_all_alerts container has only
+        platforms of interest (example: 'windows10', 'windows7', 'linux', 'osx', 'android').
+        Please filter sorted_all_alerts list with filter_alerts(alerts) before calling this function.
+        :param sorted_all_alerts: alerts sorted by platform name
+        '''
+        platform_grouped_alerts = []
+        for platform in self.ordered_platforms_of_interest:
+            specific_platform_alerts = [
+                alert for alert in sorted_all_alerts
+                if platform in alert.series_signature.platform.platform
+            ]
+            if len(specific_platform_alerts):
+                platform_grouped_alerts.append(specific_platform_alerts)
+
+        platform_picked = []
+        for alert_group in zip_longest(*platform_grouped_alerts):
+            platform_picked.extend(alert_group)
+        platform_picked = [alert for alert in platform_picked if alert is not None]
+        return platform_picked
+
+    def _os_relevance(self, alert_platform: str):
+        '''
+        One of the sorting criteria.
+        :param alert_platform: the name of the current alert's platform
+        :return: int value of platform's relevance
+        '''
+        for platform_of_interest in self.ordered_platforms_of_interest:
+            if alert_platform.startswith(platform_of_interest):
+                return len(self.ordered_platforms_of_interest) - self.ordered_platforms_of_interest.index(
+                    platform_of_interest)
+        raise ValueError('Unknown platform.')
+
+    def _has_relevant_platform(self, alert: PerformanceAlert):
+        '''
+        Filter criteria based on platform name.
+        '''
+        alert_platform_name = alert.series_signature.platform.platform
+        if any(alert_platform_name.startswith(platform_of_interest)
```

Comment:
You could just return the result of ``any`` here, I believe:

    return any(...

Reply:
Good point. Will address this in another PR of mine, as Octavian is on PTO this week.